### PR TITLE
Assessment Controller for Update

### PR DESCRIPTION
**Pull request by persas**
AssessmentController update in #1675

### CODE REVIEW

File: src/server/controller/assessment/update.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    await ActivityLogRepository.insertActivityLog(
+      {
+        activityLog: {
+          target: updatedAssessment,
+          section: 'assessment',
+          message: ActivityLogMessage.assessmentUpdate,
+          user,
+        },
+        assessment: updatedAssessment,
```
Comment (sorja) [inline src/server/controller/assessment/update.ts]: this should be `assessment` and not `updatedAssessment`

File: src/server/controller/assessment/update.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { BaseProtocol, DB } from '@server/db'
+import { AssessmentRepository } from '@server/repository/assessment/assessment'
+import { ActivityLogRepository } from '@server/repository/public/activityLog'
+
+export const update = async (
+  props: { user: User; assessment: Assessment },
+  client: BaseProtocol = DB
+): Promise<Assessment> => {
+  const { assessment, user } = props
+  // await AssessmentRepository.updateAssessmentSchema({ assessment }, client)
```
Comment (sorja) [inline src/server/controller/assessment/update.ts]: remove unused comments

File: src/server/repository/assessment/assessment/updateAssessment.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  client: BaseProtocol = DB
+): Promise<Assessment> => {
+  const { assessment } = params
+
+  const assessmentUpdated = await client.one<Assessment>(
+    `update assessment set props = $1::jsonb where id = $2 returning  *;`,
+    [JSON.stringify(assessment.props), assessment.id],
+    (assessment) => {
+      return { ...Objects.camelize(assessment), metaCache: assessment.metacache }
+    }
```
Comment (sorja) [inline src/server/repository/assessment/assessment/updateAssessment.ts]: I think this is typo, should it be `metaCache: assessment.meta_cache`?
Also, I would support updating `metaCache` like you did for props`, but @minotogna can verify if this is needed

**Comment by sorja** [comment]
Looks good in general, how did you test it was working?
We need to add integration test ;)

File: src/server/repository/assessment/assessment/updateAssessment.ts (hunk 3 above)
Comment (persas) [reply src/server/repository/assessment/assessment/updateAssessment.ts]: You are right. There is a typo, actually, it should be `assessment.metaCache`

File: src/server/controller/assessment/update.ts (hunk 1 above)
Comment (persas) [reply src/server/controller/assessment/update.ts]: Are you sure? 

In line 15 the const is defined as `updatedAssessment`. Do you mean to change the line 15 as well?

Comment (sorja) [reply src/server/controller/assessment/update.ts]: Activity log is to keep track of changes
in the assessment we pass the 'old' assessment and the 'target' is the updated one

This way when you are looking at changes, it's easy to see what changed

**Comment by minotogna** [comment]
> Looks good in general, how did you test it was working?
> 
> We need to add integration test ;)

Not now . We need to move to other tasks first

File: src/server/repository/assessment/assessment/updateAssessment.ts (hunk 3 above)
Comment (minotogna) [reply src/server/repository/assessment/assessment/updateAssessment.ts]: It should be the property of the database table

File: src/server/controller/assessment/update.ts (hunk 1 above)
Comment (minotogna) [reply src/server/controller/assessment/update.ts]: The repository method name must be renamed to update , that's what @sorja meant

Comment (sorja) [reply src/server/controller/assessment/update.ts]: > The repository method name must be renamed to update , that's what @sorja meant

Thats also a good point though; what I mean is the activity log entries

File: src/server/repository/adapter/index.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import { AssessmentAdapter } from './assessment'
+
+export const Adapter = { AssessmentAdapter }
```
Comment (minotogna) [inline src/server/repository/adapter/index.ts]: export AssessmentAdapter, do not rename it.

File: src/server/repository/assessment/assessment/updateAssessment.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+export const updateDefaultCycle = async (
+  params: {
+    assessment: Assessment
+  },
+  client: BaseProtocol = DB
+): Promise<Assessment> => {
+  const { assessment } = params
+
+  const assessmentUpdated = await client.one<Assessment>(
+    `update assessment set props = props || '{"defaultCycle": $1~}' where id = $2 returning  *;`,
```
Comment (minotogna) [inline src/server/repository/assessment/assessment/updateAssessment.ts]: we dont want to return the meta_cache here. activity log would be huge

File: src/test/integration/assessment/updateAssessmentDefaultCycle.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+      cycleName: assessmentCycleName,
+    })
+
+    const user = await UserController.getOne({
+      email: userMockTest.email,
+    })
+
+    assessment.props.defaultCycle = cycle.uuid
+
+    const updatedAssessment = await AssessmentController.updateDefaultCycle({ user, assessment })
```
Comment (minotogna) [inline src/test/integration/assessment/updateAssessmentDefaultCycle.ts]: AssessmentController.updateDefaultCycle  should take cycle as parameter: AssessmentController.updateDefaultCycle({ user, assessment, cycle }), where cycle is the cycle we want to update. 
do not set props above.

**Review by minotogna — CHANGES_REQUESTED** [review]

File: src/server/repository/assessment/assessment/updateAssessment.ts (hunk 5 above)
Comment (minotogna) [reply src/server/repository/assessment/assessment/updateAssessment.ts]: so to fix this problem, the update query should not return anything.
At the end of the method, you simply ```return read({id: assessment.id})```, where read method exist in the same folder of this file

File: src/server/repository/adapter/index.ts (hunk 4 above)
Comment (minotogna) [reply src/server/repository/adapter/index.ts]: export { AssessmentAdapter } from './assessment'

File: src/server/repository/adapter/assessment.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { Objects } from '@utils/objects'
+
+import { Assessment, AssessmentProps } from '@meta/assessment'
+
+interface AssessmentDB {
+  id: number
+  uuid: string
+  props: AssessmentProps
+  meta_cache: AssessmentProps
```
Comment (minotogna) [inline src/server/repository/adapter/assessment.ts]: meta_cache is of AssessmentMetaCache type 
import { Assessment, AssessmentMetaCache , AssessmentProps } from '@meta/assessment'

**Review by minotogna — CHANGES_REQUESTED** [review]
good pr 👍 , just a type fix AssessmentDB

**Review by minotogna — APPROVED** [review]

**Review by sorja — APPROVED** [review]

**Pull request merged**